src/panic.md

```diff
@@ -0,0 +1,127 @@
+over panic behavior:
+* A [_panic runtime_](#panic-runtimes) defined how a panic is handled during
+  runtime.
+* [FFI ABIs](items/functions.md#unwinding) may alter how panics behave.
+
+> [!NOTE]
+> The standard library provides the capability to explicitly panic
+> via the [`panic!` macro][macro-panic].
+
+## Unwinding
+
+r[panic.unwind]
+
+r[panic.unwind.intro]
+Panicking may either be recoverable or non-recoverable, though it can be
+configured (via `panic=abort`) to always be non-recoverable. (The converse is
+not true: `panic=unwind` does not guarantee that all panics are recoverable,
+only that panicking via the `panic!` macro and similar standard library
+mechanisms is recoverable.)
+
+r[panic.unwind.destruction]
+When panic recovery occurs, the runtime "unwinds" Rust frames, just as C++'s `throw` unwinds C++ frames, until the panic reaches
+the point of recovery (for instance at a thread boundary). This means that as
+the panic traverses Rust frames, live objects in those frames that [implement
+`Drop`][destructors] will have their `drop` methods called. Thus, when normal
+execution resumes, no-longer-accessible objects will have been "cleaned up"
+just as if they had gone out of scope normally.
+
+> [!NOTE]
+> As long as this guarantee of resource-cleanup is preserved,
+> "unwinding" may be implemented without actually using the mechanism used by
+> C++ for the target platform.
+
+> [!NOTE]
+> The standard library provides two mechanisms for recovering from a panic,
+> [`catch_unwind`][fn-catch-unwind] (which enables recovery within the
+> panicking thread) and [`thread::spawn`][thread-spawn] (which automatically
+> sets up panic recovery for the spawned thread so that other threads may
+> continue running).
+
+### Unwinding across FFI boundaries
+
+r[panic.unwind.ffi]
+
+r[panic.unwind.ffi.intro]
+It is possible to unwind across FFI boundaries using an [appropriate ABI
+declaration][unwind-abi]. While useful in certain cases, this creates unique
+opportunities for undefined behavior, especially when multiple language runtimes
+are involved.
+
+r[panic.unwind.ffi.undefined]
+Unwinding with the wrong ABI is undefined behavior:
+
+* Causing an unwind into Rust code from a foreign function that was called via a
+  function declaration or pointer declared with a non-unwinding ABI, such as `"C"`,
+  `"system"`, etc. (For example, this case occurs when such a function written in
+  C++ throws an exception that is uncaught and propagates to Rust.)
+* Calling a Rust `extern` function that unwinds (with `extern "C-unwind"` or
+  another ABI that permits unwinding) from a runtime that does not support.
+  unwinding, such as code compiled with GCC or Clang using `-fno-exceptions`
+
+r[panic.unwind.ffi.catch-foreign]
+Catching a foreign unwinding operation (such as a C++ exception) using
+`catch_unwind`, `JoinHandle::join`, or by letting it propagate all the way to a
+Rust `main()` function will have one of two behaviors, and it is unspecified
```

Comment:
I think to clarify here, I believe this also includes an uncaught exceptions in threads?

```suggestion
`catch_unwind`, `JoinHandle::join`, or by letting it propagate all the way to a
thread root or past a Rust `main()` function will have one of two behaviors, and it is unspecified
```